File: kill_chrome.js
const cp = require('child_process');

console.log("Fetching process list...");
cp.exec('tasklist /v /fo csv', { maxBuffer: 10 * 1024 * 1024 }, (err, stdout) => {
    if (err) {
        console.error("Error running tasklist:", err);
        return;
    }
    
    const lines = stdout.split('\n');
    let killed = 0;
    
    lines.forEach(line => {
        const lowerLine = line.toLowerCase();
        // Look for chrome.exe processes that have "Testing" in their window title or image name
        if (lowerLine.includes('chrome.exe') && lowerLine.includes('testing')) {
            // Tasklist CSV format: "Image Name","PID","Session Name","Session#","Mem Usage","Status","User Name","CPU Time","Window Title"
            const parts = line.split('","');
            if (parts.length >= 2) {
                const pid = parts[1].replace('"', '');
                if (pid && !isNaN(pid)) {
                    console.log('Killing PID ' + pid);
                    try {
                        cp.execSync('taskkill /f /pid ' + pid, { stdio: 'ignore' });
                        killed++;
                    } catch (e) {
                        console.log('Failed to kill PID ' + pid);
                    }
                }
            }
        }
    });
    
    console.log('Successfully killed ' + killed + ' Google Chrome for Testing processes.');
});

File: kill_chrome.py
import subprocess
import re

def kill_zombies():
    print("Finding Chrome processes...")
    try:
        output = subprocess.check_output('C:\\Windows\\System32\\wbem\\wmic.exe process where "name=\'chrome.exe\'" get processid,commandline', shell=True).decode('utf-8', errors='ignore')
    except Exception as e:
        print("Error getting processes:", e)
        return

    lines = output.split('\n')
    killed = 0
    for line in lines:
        if '--headless' in line.lower() or 'puppeteer' in line.lower() or 'account-' in line.lower():
            # Find PID which is at the end of the line
            parts = line.strip().split()
            if parts:
                pid = parts[-1]
                if pid.isdigit():
                    print(f"Killing PID {pid}...")
                    try:
                        subprocess.call(f'C:\\Windows\\System32\\taskkill.exe /F /PID {pid} /T', shell=True, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
                        killed += 1
                    except Exception as e:
                        print(f"Failed to kill {pid}: {e}")
    
    print(f"Done. Successfully killed {killed} headless Chrome instances.")

if __name__ == '__main__':
    kill_zombies()
